toxcore/net_crypto.c

```diff
@@ -639,6 +639,14 @@ static IP_Port return_ip_port_connection(Net_Crypto *c, int crypt_connection_id)
         return conn->ip_portv6;
     }
 
+    if (ip_is_lan(conn->ip_portv4.ip) == 0) {
+        return conn->ip_portv4;
+    }
+
+    if (net_family_is_ipv6(conn->ip_portv6.ip.family)) {
```

Comment:
I'll think about this a bit more tomorrow. Before I think about it and read more code, maybe you can tell me what the purpose of line 638-640 is after this change. I.e. why we have `ip_is_lan` between the two ipv6 returns.